common/src/main/java/org/apache/drill/common/config/DrillConfig.java

```diff
@@ -164,6 +164,15 @@ public static DrillConfig create(String overrideFileResourcePathname, boolean en
   }
 
   /**
+   * Creates a drill configuration using the provided config file.
```

Comment:
We pass Config object which contains URL to drill-module.conf in our jar. drill-module.conf contains packages that allowed to be scanned. Based on this URL, drill config is created. During creation drill finds all packages listed in drill-module.conf and adds them in drill config.
Example of created drill config:
{
    # jar:file:/home/osboxes/git_repo/drill/exec/java-exec/target/1473346938237-0/DrillUDF-1.0.jar!/drill-module.conf: 1
    "drill" : {
        # jar:file:/home/osboxes/git_repo/drill/exec/java-exec/target/1473346938237-0/DrillUDF-1.0.jar!/drill-module.conf: 1
        "classpath" : {
            # jar:file:/home/osboxes/git_repo/drill/exec/java-exec/target/1473346938237-0/DrillUDF-1.0.jar!/drill-module.conf: 1
            "scanning" : {
                # jar:file:/home/osboxes/git_repo/drill/exec/java-exec/target/1473346938237-0/DrillUDF-1.0.jar!/drill-module.conf: 1
                "packages" : [
                    # jar:file:/home/osboxes/git_repo/drill/exec/java-exec/target/1473346938237-0/DrillUDF-1.0.jar!/drill-module.conf: 1
                    "com.drill.udf"
                ]
            }
        }
    }
}